---
title: "Galaxy Gene Expression Activity"
output: html_document
---

# Overview of activity and data

Our Galaxy activity is a condensed tutorial based on the ["Reference-based RNA-Seq data analysis"](https://training.galaxyproject.org/training-material/topics/transcriptomics/tutorials/ref-based/tutorial.html) Galaxy Training Tutorial. 

It uses [data that is deposited on and available from zenodo](https://zenodo.org/records/6457007), including subsampled data that will be quicker to work with. For more info on the data, checkout the tutorial linked above.

# Activity steps

## Set up Galaxy's history pane

- [ ]  If you have files in your history already, use the plus sign button on the top right of the history pane to `Create new history`.

- [ ]  Click the pencil button next to "Unnamed history". Fill in the name with something descriptive/appropriate and add more detail a description to the annotation if you want. Click "Save"

Our History pane is empty and we'll need to load data. 

## Data upload

Why do we want sequencing reads and a reference genome? Why are there 4 files for sequencing reads? 

### Sequencing reads

- [ ]  Copy these links:

```
https://zenodo.org/record/6457007/files/GSM461177_1_subsampled.fastqsanger
https://zenodo.org/record/6457007/files/GSM461177_2_subsampled.fastqsanger
https://zenodo.org/record/6457007/files/GSM461180_1_subsampled.fastqsanger
https://zenodo.org/record/6457007/files/GSM461180_2_subsampled.fastqsanger
```

- [ ]  In Galaxy, click the "Upload" button in the top left of the page. 

```{r echo=FALSE, fig.alt="In the top left of the page, there is a button named Upload with a rectangle and an arrow pointing up. Please click that button"}
ottrpal::include_slide("https://docs.google.com/presentation/d/1kWsS23lOJxfbhE8jSdE92JWnEceUEYm5xovCczPbe-8/edit#slide=id.g281646704fe_0_0")
```

This will open up an interactive panel for data upload:

```{r echo=FALSE, fig.alt="This is the interactive panel that opens for data upload"}
ottrpal::include_slide("https://docs.google.com/presentation/d/1kWsS23lOJxfbhE8jSdE92JWnEceUEYm5xovCczPbe-8/edit#slide=id.g2f3201278fa_0_0")
```

- [ ]  Click the "Paste/Fetch Data" button in the middle of the bottom stretch of options.

```{r echo=FALSE, fig.alt="Please click the Paste/Fetch data button in the middle of the bottom stretch of the "}
ottrpal::include_slide("https://docs.google.com/presentation/d/1kWsS23lOJxfbhE8jSdE92JWnEceUEYm5xovCczPbe-8/edit#slide=id.g281646704fe_0_5")
```

- [ ]  Paste the copied URLs into the middle box. 

```{r echo=FALSE}
ottrpal::include_slide("https://docs.google.com/presentation/d/1kWsS23lOJxfbhE8jSdE92JWnEceUEYm5xovCczPbe-8/edit#slide=id.g281646704fe_0_10")
```

- [ ]  Using the first dropdown menu on the top (labeled "Auto-detect"), let's select the filetype: `fastqsanger` (Note the list includes both `fastqcsanger` and `fastqsanger` where one is QC and the other is just q. Select the one with just a q). 

```{r echo=FALSE}
ottrpal::include_slide("https://docs.google.com/presentation/d/1kWsS23lOJxfbhE8jSdE92JWnEceUEYm5xovCczPbe-8/edit#slide=id.g281646704fe_0_16")
```

- [ ]  Using the second dropdown menu on the top (labeled "unspecified (?)"), let's select the reference organism: `D. melanogaster Aug. 2014 (BDGP Release 6 + ISO1 MT/dm6) (dm6)`

```{r echo=FALSE}
ottrpal::include_slide("https://docs.google.com/presentation/d/1kWsS23lOJxfbhE8jSdE92JWnEceUEYm5xovCczPbe-8/edit#slide=id.g281646704fe_0_21")
```

- [ ]  Click the blue "Start" button in the bottom stretch of options. 

```{r echo=FALSE}
ottrpal::include_slide("https://docs.google.com/presentation/d/1kWsS23lOJxfbhE8jSdE92JWnEceUEYm5xovCczPbe-8/edit#slide=id.g281646704fe_0_27")
```

- [ ]  Click the "Close" button at the end of the bottom stretch of options. 

```{r echo=FALSE}
ottrpal::include_slide("https://docs.google.com/presentation/d/1kWsS23lOJxfbhE8jSdE92JWnEceUEYm5xovCczPbe-8/edit#slide=id.g281646704fe_0_33")
```

#### Creating a paired collection

- [ ]  Click the "Select items" check in a box button on the left of the banner above the listed datasets

```{r echo=FALSE}
ottrpal::include_slide("https://docs.google.com/presentation/d/1kWsS23lOJxfbhE8jSdE92JWnEceUEYm5xovCczPbe-8/edit#slide=id.g2f3201278fa_0_6")
```

- [ ]  Click "Select all" that appears on the right of the banner

```{r echo=FALSE}
ottrpal::include_slide("https://docs.google.com/presentation/d/1kWsS23lOJxfbhE8jSdE92JWnEceUEYm5xovCczPbe-8/edit#slide=id.g2f3201278fa_0_12")
```

- [ ]  Click the down arrow

```{r echo=FALSE}
ottrpal::include_slide("https://docs.google.com/presentation/d/1kWsS23lOJxfbhE8jSdE92JWnEceUEYm5xovCczPbe-8/edit#slide=id.g2f3201278fa_0_16")
```

- [ ]  Click "Build List of Dataset Pairs".

```{r echo=FALSE}
ottrpal::include_slide("https://docs.google.com/presentation/d/1kWsS23lOJxfbhE8jSdE92JWnEceUEYm5xovCczPbe-8/edit#slide=id.g2f3201278fa_0_20")
```

This will open up an interactive panel: 

```{r echo=FALSE}
ottrpal::include_slide("https://docs.google.com/presentation/d/1kWsS23lOJxfbhE8jSdE92JWnEceUEYm5xovCczPbe-8/edit#slide=id.g2f3201278fa_0_24")
```

- [ ]  In the bottom right corner, enter `2 PE fastqs` as the name

```{r echo=FALSE}
ottrpal::include_slide("https://docs.google.com/presentation/d/1kWsS23lOJxfbhE8jSdE92JWnEceUEYm5xovCczPbe-8/edit#slide=id.g281646704fe_0_43")
```

- [ ]  In the green strips, there are 3 columns, for each fastqsanger pair, in the middle column we'll edit the displayed name to be a more informative name. 

  - [ ]  Click on "GSM461177_subsampled", and enter "GSM461177_untreat_paired"
  - [ ]  Click on "GSM461180_subsampled", and enter "GSM461180_treat_paired"

```{r echo=FALSE}
ottrpal::include_slide("https://docs.google.com/presentation/d/1kWsS23lOJxfbhE8jSdE92JWnEceUEYm5xovCczPbe-8/edit#slide=id.g281646704fe_0_48")
```

- [ ]  Click the blue "Create collection" button on the bottom right

### Reference genome annotation

- [ ]  Copy this link:

```
https://zenodo.org/record/6457007/files/Drosophila_melanogaster.BDGP6.32.109_UCSC.gtf.gz
```

- [ ]  In Galaxy, click the "Upload" button in the top left of the page. This will open up an interactive panel for data upload.

```{r echo=FALSE, fig.alt="In the top left of the page, there is a button named Upload with a rectangle and an arrow pointing up. Please click that button"}
ottrpal::include_slide("https://docs.google.com/presentation/d/1kWsS23lOJxfbhE8jSdE92JWnEceUEYm5xovCczPbe-8/edit#slide=id.g281646704fe_0_0")
```

This will open up an interactive panel for data upload:

```{r echo=FALSE, fig.alt="This is the interactive panel that opens for data upload"}
ottrpal::include_slide("https://docs.google.com/presentation/d/1kWsS23lOJxfbhE8jSdE92JWnEceUEYm5xovCczPbe-8/edit#slide=id.g2f3201278fa_0_28")
```

- [ ]  Click the "Paste/Fetch Data" button in the middle of the bottom stretch of options.

```{r}
ottrpal::include_slide("https://docs.google.com/presentation/d/1kWsS23lOJxfbhE8jSdE92JWnEceUEYm5xovCczPbe-8/edit#slide=id.g281646704fe_0_59")
```

- [ ]  Paste the copied URL into the middle box.

- [ ]  Using the first dropdown menu on the top (labeled "Auto-detect"), let's select the filetype: `gtf`. 

- [ ]  Using the second dropdown menu on the top (labeled "unspecified (?)"), let's select the reference organism: `D. melanogaster Aug. 2014 (BDGP Release 6 + ISO1 MT/dm6) (dm6)`

- [ ]  Click the blue "Start" button in the bottom stretch of options. 

- [ ]  Click the "Close" button at the end of the bottom stretch of options.

## Quality Control

Now that we have all of the data uploaded, we'll begin with some quality control analysis of the data. This is useful for verifying that the data is high quality, but also will benefit us when we run later steps/need to know info as inputs for the mapping tools (e.g., read size).

### FastQC

- [ ]  On the top left of the page, the tool pane has a search bar. Type `Flatten` into the search bar and select the `Flatten collection` tool. This will open the Flatten collection tool in the middle pane.

```{r echo=FALSE}
ottrpal::include_slide("https://docs.google.com/presentation/d/1kWsS23lOJxfbhE8jSdE92JWnEceUEYm5xovCczPbe-8/edit#slide=id.g2f3201278fa_0_35")
```

- [ ] In the middle pane, if the "Input Collection:" is not already filled in with "2 PE fastqs", click the down arrow and select it. 

```{r echo=FALSE}
ottrpal::include_slide("https://docs.google.com/presentation/d/1kWsS23lOJxfbhE8jSdE92JWnEceUEYm5xovCczPbe-8/edit#slide=id.g2f3201278fa_0_39")
```

- [ ] Click the blue "> Run Tool" button. This will add the job to the queue and add the output to the top of the history pane.

```{r echo=FALSE}
ottrpal::include_slide("https://docs.google.com/presentation/d/1kWsS23lOJxfbhE8jSdE92JWnEceUEYm5xovCczPbe-8/edit#slide=id.g281646704fe_0_65")
```

- [ ] You can rename the output to a more informative name by 

  - [ ]  Clicking the pencil (the middle icon) for that dataset in the history pane. The icons are on the right side.
  
```{r echo=FALSE}
ottrpal::include_slide("https://docs.google.com/presentation/d/1kWsS23lOJxfbhE8jSdE92JWnEceUEYm5xovCczPbe-8/edit#slide=id.g2f3201278fa_0_43")
```
  
  - [ ]  In the "Edit Collection Attributes" pane that opens in the middle panel, enter a more informative "Name"

```{r echo=FALSE}
ottrpal::include_slide('https://docs.google.com/presentation/d/1kWsS23lOJxfbhE8jSdE92JWnEceUEYm5xovCczPbe-8/edit#slide=id.g2f3201278fa_0_47')
```
  
  - [ ]  Click the blue "Save" button

- [ ] On the top left of the page, using the tool pane search bar, type `Fastq` into the search bar and select the `FastQC` tool. This will open the FastQC tool in the middle panel.

```{r echo=FALSE}
ottrpal::include_slide("https://docs.google.com/presentation/d/1kWsS23lOJxfbhE8jSdE92JWnEceUEYm5xovCczPbe-8/edit#slide=id.g281646704fe_0_73")
```
 
- [ ] In the blue banner highlighted section, select the file folder "Dataset collection" icon

```{r echo=FALSE}
ottrpal::include_slide("https://docs.google.com/presentation/d/1kWsS23lOJxfbhE8jSdE92JWnEceUEYm5xovCczPbe-8/edit#slide=id.g281646704fe_0_77")
```

- [ ] If the "Raw read data" from your current history doesn't automatically fill with the renamed output from the Flatten collection tool, select that dataset as input. 

```{r echo=FALSE}
ottrpal::include_slide("https://docs.google.com/presentation/d/1kWsS23lOJxfbhE8jSdE92JWnEceUEYm5xovCczPbe-8/edit#slide=id.g281646704fe_0_82")
```
 
### MultiQC to combine FASTQC output
 
- [ ] On the top left of the page, using the tool pane search bar, type `multi` into the search bar and select the `MultiQC` tool. This will open the `MultiQC` tool in the middle pane.  

```{r echo=FALSE}
ottrpal::include_slide("https://docs.google.com/presentation/d/1kWsS23lOJxfbhE8jSdE92JWnEceUEYm5xovCczPbe-8/edit#slide=id.g281646704fe_0_88")
```

```{r echo=FALSE}
ottrpal::include_slide("https://docs.google.com/presentation/d/1kWsS23lOJxfbhE8jSdE92JWnEceUEYm5xovCczPbe-8/edit#slide=id.g281646704fe_0_93")
```

- [ ] Within the Results section, for the `Which tool was used to generate logs?` question, use the down arrow to see a list and scroll down until you see `FastQC` and select `FastQC`.

```{r echo=FALSE}
ottrpal::include_slide("https://docs.google.com/presentation/d/1kWsS23lOJxfbhE8jSdE92JWnEceUEYm5xovCczPbe-8/edit#slide=id.g281646704fe_0_97")
```

- [ ] In the FastQC output section, click the `+ Insert FastQC output` button. 

```{r echo=FALSE}
ottrpal::include_slide("https://docs.google.com/presentation/d/1kWsS23lOJxfbhE8jSdE92JWnEceUEYm5xovCczPbe-8/edit#slide=id.g281646704fe_0_101")
```

- [ ] In the blue banner highlighted section, select the file folder "Dataset collection" icon

```{r echo=FALSE}
ottrpal::include_slide("https://docs.google.com/presentation/d/1kWsS23lOJxfbhE8jSdE92JWnEceUEYm5xovCczPbe-8/edit#slide=id.g281646704fe_0_104")
```
 
- [ ] Then with the down arrow, select the `FASTQC on collection __: RawData` data set
 
```{r echo=FALSE}
ottrpal::include_slide("https://docs.google.com/presentation/d/1kWsS23lOJxfbhE8jSdE92JWnEceUEYm5xovCczPbe-8/edit#slide=id.g281646704fe_0_112") 
```
  
- [ ] Optionally, you can add a Report title near the bottom of the middle pane
 
- [ ] Click the blue Run tool button in the upper right of the middle pane
 
```{r echo=FALSE}
ottrpal::include_slide("https://docs.google.com/presentation/d/1kWsS23lOJxfbhE8jSdE92JWnEceUEYm5xovCczPbe-8/edit#slide=id.g281646704fe_0_116")
```

- [ ] Let's open and inspect the webpage output at the top of the history pane. To view the output file, click the eye icon. To download the output, click the save/floppy disc icon. 
 
```{r echo=FALSE}
ottrpal::include_slide("https://docs.google.com/presentation/d/1kWsS23lOJxfbhE8jSdE92JWnEceUEYm5xovCczPbe-8/edit#slide=id.g281646704fe_0_126")
```
 
### Cutadapt / Trim adaptors

- [ ] On the top left of the page, using the tool pane search bar, type `Cut` into the search bar and select the `Cutadapt` tool. This will open the `Cutadapt` tool in the middle pane.  
  
```{r echo=FALSE}
ottrpal::include_slide("https://docs.google.com/presentation/d/1kWsS23lOJxfbhE8jSdE92JWnEceUEYm5xovCczPbe-8/edit#slide=id.g281646704fe_0_121")
```

```{r echo=FALSE}
ottrpal::include_slide("https://docs.google.com/presentation/d/1kWsS23lOJxfbhE8jSdE92JWnEceUEYm5xovCczPbe-8/edit#slide=id.g281646704fe_0_133")
```

- [ ]  For `Single-end or Paired-end reads?` click the down arrow and select `Paired-end Collection`. 
  
```{r echo=FALSE}
ottrpal::include_slide("https://docs.google.com/presentation/d/1kWsS23lOJxfbhE8jSdE92JWnEceUEYm5xovCczPbe-8/edit#slide=id.g281646704fe_0_137")
```
  
- [ ]  Verify that it selected your `2 PE fastqs` as the paired collection input, if not, select it.
  
```{r echo=FALSE}
ottrpal::include_slide("https://docs.google.com/presentation/d/1kWsS23lOJxfbhE8jSdE92JWnEceUEYm5xovCczPbe-8/edit#slide=id.g281646704fe_0_141")
```

- [ ] Scroll down to the `Other Read Trimming Options` section and edit the `Quality cutoff(s) (R1)*` parameter. Enter a value of 20.
  
```{r echo=FALSE}
ottrpal::include_slide("https://docs.google.com/presentation/d/1kWsS23lOJxfbhE8jSdE92JWnEceUEYm5xovCczPbe-8/edit#slide=id.g281646704fe_0_146")
```

- [ ]  Scroll down to the `Read Filtering Options` section and edit the `Minimum length (R1)` parameter. Enter a value of 20.
  
```{r echo=FALSE}
ottrpal::include_slide("https://docs.google.com/presentation/d/1kWsS23lOJxfbhE8jSdE92JWnEceUEYm5xovCczPbe-8/edit#slide=id.g281646704fe_0_156")
```

- [ ] Scroll down to the `Additional outputs to generate` checkbox section and check the `Report: Cutadapt's per-adapter statistics. You can use this file with MultiQC`

```{r echo=FALSE}
ottrpal::include_slide("https://docs.google.com/presentation/d/1kWsS23lOJxfbhE8jSdE92JWnEceUEYm5xovCczPbe-8/edit#slide=id.g281646704fe_0_160")
```
  
- [ ] Click the blue Run tool button

### View Cutadapt results with MultiQC

- [ ] On the top left of the page, using the tool pane search bar, type `multi` into the search bar and select the `MultiQC` tool. This will open the `MultiQC` tool in the middle pane.  

```{r echo=FALSE}
ottrpal::include_slide("https://docs.google.com/presentation/d/1kWsS23lOJxfbhE8jSdE92JWnEceUEYm5xovCczPbe-8/edit#slide=id.g281646704fe_0_88")
```

```{r echo=FALSE}
ottrpal::include_slide("https://docs.google.com/presentation/d/1kWsS23lOJxfbhE8jSdE92JWnEceUEYm5xovCczPbe-8/edit#slide=id.g281646704fe_0_93")
```

- [ ] Within the `Results` section and `Which tool was used to generate logs` subsection, click the down arrow and select `Cutadapt/Trim Galore!`. 

```{r echo=FALSE}
ottrpal::include_slide("https://docs.google.com/presentation/d/1kWsS23lOJxfbhE8jSdE92JWnEceUEYm5xovCczPbe-8/edit#slide=id.g281646704fe_0_168")

```

- [ ] In the blue banner highlighted section, select the file folder "Dataset collection" icon & then with the down arrow, select the `Cutadapt on collection __: Report` data set 

```{r echo=FALSE}
ottrpal::include_slide("https://docs.google.com/presentation/d/1kWsS23lOJxfbhE8jSdE92JWnEceUEYm5xovCczPbe-8/edit#slide=id.g281646704fe_0_172")
```

- [ ] We can explore this output as well to see how much of the data was trimmed

## Next steps: Mapping with RNA STAR

Follow the steps in the [Galaxy walkthrough to continue with mapping](https://training.galaxyproject.org/training-material/topics/transcriptomics/tutorials/ref-based/tutorial.html#mapping)

## Additional Resources 

Here are some other relevant tutorials from Galaxy:

### Bulk-RNA Seq 

[RNA-Seq Reads to Counts](https://usegalaxy.org/training-material/topics/transcriptomics/tutorials/rna-seq-reads-to-counts/tutorial.html#1-rna-seq-reads-to-counts)

[RNA-Seq Counts to Genes (differentially expressed genes)](https://usegalaxy.org/training-material/topics/transcriptomics/tutorials/rna-seq-counts-to-genes/tutorial.html)

[RNA-Seq Genes to Pathways (GSEA)](https://usegalaxy.org/training-material/topics/transcriptomics/tutorials/rna-seq-genes-to-pathways/tutorial.html)

[Tutorials for Visualizing RNA-seq Results](https://usegalaxy.org/training-material/topics/transcriptomics/tutorials/rna-seq-genes-to-pathways/tutorial.html)