---
title: "Galaxy Gene Expression Activity"
output: html_document
---

# Overview of activity and data

Our Galaxy activity is a condensed tutorial based on the ["Reference-based RNA-Seq data analysis"](https://training.galaxyproject.org/training-material/topics/transcriptomics/tutorials/ref-based/tutorial.html) Galaxy Training Tutorial. 

It uses [data that is deposited on and available from zenodo](https://zenodo.org/records/6457007), including subsampled data that will be quicker to work with. 

# Activity steps

## Set up Galaxy's history pane

- [ ]  If you have files in your history already, use the plus sign button on the top right of the history pane to `Create new history`.

- [ ]  Click the pencil button next to "Unnamed history". Fill in the name with something descriptive/appropriate and add more detail a description to the annotation if you want. Click "Save"

Our History pane is empty and we'll need to load data. 

## Data upload

Why do we want sequencing reads and a reference genome? Why are there 4 files for sequencing reads? 

### Sequencing reads

- [ ]  Copy these links:

```
https://zenodo.org/record/6457007/files/GSM461177_1_subsampled.fastqsanger
https://zenodo.org/record/6457007/files/GSM461177_2_subsampled.fastqsanger
https://zenodo.org/record/6457007/files/GSM461180_1_subsampled.fastqsanger
https://zenodo.org/record/6457007/files/GSM461180_2_subsampled.fastqsanger
```

- [ ]  In Galaxy, click the "Upload" button in the top left of the page. This will open up an interactive panel for data upload.

#ottrpal::include_slide("ADD SCREENSHOT HERE")

- [ ]  Click the "Paste/Fetch Data" button in the middle of the bottom stretch of options.

- [ ]  Paste the copied URLs into the middle box. 

- [ ]  Using the first dropdown menu on the top (labeled "Auto-detect"), let's select the filetype: `fastqsanger` (Note the list includes both `fastqcsanger` and `fastqsanger` where one is QC and the other is just q. Select the one with just a q). 

- [ ]  Using the second dropdown menu on the top (labeled "unspecified (?)"), let's select the reference organism: `D. melanogaster Aug. 2014 (BDGP Release 6 + ISO1 MT/dm6) (dm6)`

- [ ]  Click the blue "Start" button in the bottom stretch of options. 

- [ ]  Click the "Close" button at the end of the bottom stretch of options. 

#### Creating a paired collection

- [ ]  Click the "Select items" checkbox on the left of the banner above the listed datasets

- [ ]  Click "Select all" 

- [ ]  Click the down arrow

- [ ]  Click "Build a List of Dataset Pairs". This will open up an interactive panel 

- [ ]  In the bottom right corner, enter `2 PE fastqs` as the name

- [ ]  In the green strips, there are 3 columns, for each fastqsanger pair, in the middle column we'll edit the displayed name to be a more informative name. 

  - [ ]  Click on "GSM461177_subsampled", and enter "GSM461177_untreat_paired"
  - [ ]  Click on "GSM461180_subsampled", and enter "GSM461180_treat_paired"

- [ ]  Click the blue "Create collection" button on the bottom right

### Reference genome annotation

- [ ]  Copy this link:

```
https://zenodo.org/record/6457007/files/Drosophila_melanogaster.BDGP6.32.109_UCSC.gtf.gz
```

- [ ]  In Galaxy, click the "Upload" button in the top left of the page. This will open up an interactive panel for data upload.

#ottrpal::include_slide("ADD SCREENSHOT HERE")

- [ ]  Click the "Paste/Fetch Data" button in the middle of the bottom stretch of options.

- [ ]  Paste the copied URL into the middle box.

- [ ]  Using the first dropdown menu on the top (labeled "Auto-detect"), let's select the filetype: `gtf`. 

- [ ]  Using the second dropdown menu on the top (labeled "unspecified (?)"), let's select the reference organism: `D. melanogaster Aug. 2014 (BDGP Release 6 + ISO1 MT/dm6) (dm6)`

- [ ]  Click the blue "Start" button in the bottom stretch of options. 

- [ ]  Click the "Close" button at the end of the bottom stretch of options.

## Quality Control

Now that we have all of the data uploaded, we'll begin with some quality control analysis of the data. This is useful for verifying that the data is high quality, but also will benefit us when we run later steps/need to know info as inputs for the mapping tools.

### FastQC

- [ ]  On the top left of the page, the tool pane has a search bar. Type `Flatten` into the search bar and select the `Flatten collection` tool. This will open the Flatten collection tool in the middle pane.

- [ ] If the "Input Collection:" is not already filled in with "2 PE fastqs", click the down arrow and select it. 

- [ ] Click the blue "> Run Tool" button. This will add the job to the queue and add the output to the top of the history pane.

- [ ] You can rename the output to a more informative name by 

  - [ ]  Clicking the pencil (the middle icon) for that dataset in the history pane. The icons are on the right side.
  
  - [ ]  In the "Edit Collection Attributes" pane that opens in the middle panel, enter a more informative "Name"
  
  - [ ]  Click the blue "Save" button

### MultiQC to combine FASTQC output

### Cutadapt / Trim adaptors

## Mapping

### Run STAR

### MultiQC to combine log output from STAR 

